src/btree/bt_curnext.c

```diff
@@ -792,6 +795,22 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
 skip_page:
         if (F_ISSET(cbt, WT_CBT_READ_ONCE))
             LF_SET(WT_READ_WONT_NEED);
+
+#ifdef HAVE_DIAGNOSTIC
+        if (cbt->last_ref != NULL)
+            WT_ERR(__wt_page_release(session, cbt->last_ref, 0));
+
+        cbt->last_ref = cbt->ref;
+        if (cbt->last_ref) {
+            WT_ERR(__wt_hazard_set(session, cbt->last_ref, &busy));
+            // Loop until we get an hazard pointer.
+            while (busy) {
```

Comment:
Should we exit at some point if busy stays `true` ?

Reply:
I have removed the loop.